spec/definitions.yaml

```diff
@@ -788,3 +795,268 @@ definitions:
+        properties:
+          min:
+            description: Minimum number of nodes in the pool
+            type: integer
+          max:
+            description: Maximum number of nodes in the pool
+            type: integer
+      node_spec:
+        description: Worker node specification
+        type: object
+        $ref: '#/definitions/V5AddNodePoolRequestNodeSpec'
+
+  # Specification of a node in a node pool to be created
+  V5AddNodePoolRequestNodeSpec:
+    type: object
+    properties:
+      aws:
+        type: object
+        description: |
+          Attributes specific to the AWS provider
+        properties:
+          instance_type:
+            type: string
+            description: |
+              EC2 instance type to use for all nodes in the node pool
+
+  # Specification of a node in a node pool after creation
+  V5GetNodePoolResponseNodeSpec:
+    type: object
+    properties:
+      aws:
+        type: object
+        description: |
+          Attributes specific to the AWS provider
+        properties:
+          instance_type:
+            type: string
+            description: |
+              EC2 instance type used by all nodes in this pool
+      volume_sizes_gb:
+        type: object
+        properties:
+          docker:
+            description: |
+              Size of the Docker volume, used to store docker images and
+              overlay file system data, in GB.
+            type: integer
+          kubelet:
+            description: |
+              Size of the kubelet volume of the worker, used e. g. for
+              volumes of type EmptyDir, in GB.
+            type: integer
+
+  V5GetNodePoolResponse:
+    type: object
+    properties:
+      id:
+        description: Node pool identifier. Unique within a tenant cluster.
+        type: string
+      name:
+        description: Node pool name
+        type: string
+      availability_zones:
+        description: |
+          Names of the availability zones used by the nodes of this pool.
+        type: array
+        items:
+          description: Name of the availability zone
+          type: string
+      scaling:
+        description: |
+          Scaling range of the node pool.
+          If `min` and `max` differ, the node pool is scaled automatically.
+          If both have the same value, the cluster size is pinned.
+        type: object
+        properties:
+          min:
+            description: Minimum number of nodes in the pool
+            type: integer
+          max:
+            description: Maximum number of nodes in the pool
+            type: integer
+      subnet:
+        description: IP address block used by the node pool
+        type: string
+      node_spec:
+        description: Worker node specification
+        type: object
+        $ref: '#/definitions/V5GetNodePoolResponseNodeSpec'
+
+
+  # Request to create a new cluster in V5
+  V5AddClusterRequest:
+    type: object
+    required:
+      - owner
+    properties:
```

Comment:
Do we yet know which fields are required for creation?

Reply:
Ideally none should be required and we would have a sensible default for any of them.